### PR TITLE
DAOS-7244 dtx: more info when fail to locate target for dtx RPC

### PR DESCRIPTION
**Pull request by Nasf-Fan**
Add some log messages if fail to locate the target for DTX RPC.

The patch also replaces write lock on pool map as read lock when
DTX logic accesses pool map, that will reduce lock contention.

Signed-off-by: Fan Yong <fan.yong@intel.com>

### CODE REVIEW

File: src/dtx/dtx_rpc.c [hunk 1]
```diff
@@ -869,7 +877,9 @@ dtx_refresh_internal(struct ds_cont_child *cont, int *check_count,
 		ABT_rwlock_wrlock(pool->sp_lock);
```
Comment (NiuYawei) [inline src/dtx/dtx_rpc.c:877]: It's not from this patch, but I'm wondering why a write lock is required here? When the batched commit/refresh ULTs are running concurrently on VOS xstreams, I think the lock contention will hurt performance badly.

Comment (Nasf-Fan) [reply src/dtx/dtx_rpc.c]: Ah, it should be rd lock. I will refresh the patch to fix that.

**Review by NiuYawei — DISMISSED** [review]

**Comment by Nasf-Fan** [comment]
Refresh the patch because of conflict with the latest master.

**Review by jolivier23 — APPROVED** [review]
should be be asserting in such cases?

**Review by NiuYawei — APPROVED** [review]

**Comment by Nasf-Fan** [comment]
It is replaced by the patch https://github.com/daos-stack/daos/pull/6006

**Pull request closed without merging**